packages/moda-sdk/src/profile/updateProfileFor.test.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  contractAddress,
+  metadataUri,
+  profileAddress,
+} from './testing/constants.ts';
+import { mockConfig } from './testing/mockConfig.ts';
+import { updateProfileFor } from './updateProfileFor.ts';
+
+describe('updateProfileFor', () => {
+  it('Returns a txHash', async () => {
+    const result = await updateProfileFor({
+      config: mockConfig,
+      profileAddress,
+      contractAddress,
+      metadataUri,
+    });
+
+    expect(result.ok).toBe(true);
+    if (result.ok) {
+      expect(result.value).toBe('txHash');
+    } else {
+      fail('Expected result to be successful');
+    }
+  });
+
+  it('Should fail if config is not verifiable', async () => {
+    const result = await updateProfileFor({
+      config: { publicClient: undefined, walletClient: undefined },
+      profileAddress,
+      contractAddress,
+      metadataUri,
+    });
+
+    expect(result.ok).toBe(false);
```

Comment:
redundant

Let's also test for when public client is valid and wallet client is not:

`const invalidWalletClient = { ...mockConfig, walletClient: undefined }`